src/berry_mill/builder.py

```diff
@@ -0,0 +1,198 @@
+                self._fcleanbox = True
+                os.makedirs(boxdir)
+
+            self._boxtmpkeydir:str = tempfile.mkdtemp(prefix="berrymill-keys-", dir= self._boxrootdir)
+
+            self._boxtmpargdir:str = tempfile.mkdtemp(prefix="berrymill-args-", dir= self._boxrootdir)
+
+
+    def _get_relative_file_uri(self, repo_key_path) -> str:
+        """
+        Change the file:// URIs in the repo key values to be relative to the boxroot
+        """
+        if not self._boxtmpkeydir:
+            raise Exception("Key directory not available")
+        if not repo_key_path:
+            raise Exception("Key path not defined")
+        return "file:///" + \
+               quote( \
+               os.path.join( \
+               os.path.basename(self._boxtmpkeydir), os.path.basename(repo_key_path)))
+
+
+    def _write_repokeys_box(self, repos:Dict[str, Dict[str, str]]) -> None:
+        """
+        Write repo keys from the self._tmp dir to the boxroot, so kiwi box can access the keys
+        It expects a file:// uri to be present in repos[reponame][key]
+        """
+        dst = self._boxtmpkeydir
+        if not dst:
+            raise Exception("ERROR: Boxroot directory is not defined")
+
+        for reponame in repos:
+            k = repos.get(reponame, {}).get("key")
+            parsed_url = urlparse(k)
+            try:
+                shutil.copy(parsed_url.path, dst)
+            except Exception as exc:
+                print(f"ERROR: Failure while trying to copying the keyfile at {parsed_url.path}")
+                print(exc)
+                return
+            repos.get(reponame, {})["key"] = self._get_relative_file_uri(parsed_url.path)
+
+    def process(self) -> None:
+        """
+        Run builder. It supposed to be already within that directory (os.chdir).
+        Directory is changed by the parent caller of the KiwiBuilder class.
+        """
+        # options that are solely accepted by kiwi-ng
+        kiwi_options = self._kiwi_options
+        # options, solely accepted by box-build plugin
+        box_options:List[str] = ["--box","ubuntu"]
+
+        if self._kiwiparams.get("debug", False):
+            box_options.append("--box-debug")
+
+        if self._params.get("cpu"):
+            box_options = ["--cpu", self._params.get("cpu")] + box_options
+
+        if self._params.get("box_memory"):
+            box_options += ["--box-memory", self._params.get("box_memory")]
+
+        if machine() == "aarch64":
+            box_options += ["--machine", "virt"]
+
+        allow_no_accel:bool = True
+        # TODO: When using cross, e.g. cpu param needs to be disabled
+        if self._params.get("cross") and machine() == "x86_64":
+            box_options += ["--aarch64", "--cpu", "cortex-a57", "--machine", "virt", "--no-accel"]
+            kiwi_options += ["--target-arch", "aarch64"]
+            allow_no_accel = False
+
+        if self._params.get("accel", False) and allow_no_accel:
```

Comment:
That is a bit unclear: `.get` returns a key, which can have `False` value, or returns `None` if there is no key. It would be much better to narrow it down:

```python
if "accel" in self._params:
```

Or:

```python
if self._params.get("accel") is False
```

Reply:
well that was not really the intention...
I changed the parameter to be named "no_accel"